edx_adapt/data/course_repository.py

```diff
@@ -82,14 +80,10 @@ def _add_problem(self, course_id, skill_names, problem_name, tutor_url, b_pretes
             }
         )
 
-    def post_problem(self, course_id, skill_names, problem_name, tutor_url):
-        self._add_problem(course_id, skill_names, problem_name, tutor_url, False, False)
-
-    def post_pretest_problem(self, course_id, skill_names, problem_name, tutor_url):
-        self._add_problem(course_id, skill_names, problem_name, tutor_url, True, False)
-
-    def post_posttest_problem(self, course_id, skill_names, problem_name, tutor_url):
-        self._add_problem(course_id, skill_names, problem_name, tutor_url, False, True)
+    def post_problem(self, course_id, skill_names, problem_name, tutor_url, pretest=None, posttest=None):
+        pretest = pretest or False
+        posttest = posttest or False
```

Comment:
Make default values `pretest=False` & `posttest=False` in function signature and drop these two lines.